realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -57,6 +59,8 @@ JNIEXPORT void JNICALL Java_io_realm_internal_SharedRealm_nativeInit(JNIEnv* env
 {
     TR_ENTER()
 
+    hack_init();
```

Comment:
I think using `__attribute__((constructor));` is also an option, if I remember it runs when the `.so` is loaded https://github.com/realm/realm-js/blob/master/src/android/jsc_override.cpp#L38